pkg/agent/controller/traceflow/traceflow_controller.go

```diff
@@ -299,10 +327,18 @@ func (c *Controller) injectPacket(tf *opsv1alpha1.Traceflow) error {
 			dstIP = dstPod.Status.PodIP
 			dstNodeIP = dstPod.Status.HostIP
 		}
+	} else if tf.Spec.Destination.Service != "" {
+		dstSvc, err := c.serviceLister.Services(tf.Spec.Destination.Namespace).Get(tf.Spec.Destination.Service)
+		if err != nil {
+			return err
+		}
+		dstIP = dstSvc.Spec.ClusterIP
 	}
-	if dstNodeIP != "" {
+	// Check encap status if Node of destination Pod is remote or destination is IP/Service
+	if dstNodeIP != "" || tf.Spec.Destination.Pod == "" {
```

Comment:
Doesn't it need to wait for a while if the destination is service? what if the target pod is on another Node? so do IP on another Node case.

Reply:
Yes. If destination is Service or IP/Pod not on current Node, then verify the tunnel/encap status and wait.
Change
`if dstNodeIP != "" || tf.Spec.Destination.Pod == "" {`
To
`if dstMAC == "" {`
And added comments.